domain/application/modelmigration/export.go

```diff
@@ -45,6 +47,13 @@ type ExportService interface {
 	GetCharm(ctx context.Context, id corecharm.ID) (internalcharm.Charm, charm.CharmOrigin, error)
 }
 
+// NotImplementedDeleteSecretState defines a secret state which does nothing.
```

Comment:
Is this temporary or permanent? It makes sense for this to be permanent, so this comment should have some substance as to why you wouldn't want any secrets to be deleted during import/export...

I'd also choose a better name like `NoopDeleteSecretState`. Words like *Not Implemented* and *Unimplemented* almost imply that they should at some point be implemented.